app-frontend/src/app/services/projects/project.service.js

```diff
@@ -148,6 +148,55 @@ export default (app) => {
             return this.Project.query(params).$promise;
         }
 
+        searchQuery() {
+            let deferred = this.$q.defer();
+            let pageSize = 1000;
+            let firstPageParams = {
+                pageSize: pageSize,
+                page: 0,
+                sort: 'createdAt,desc'
+            };
+
+            let firstRequest = this.query(firstPageParams);
+
+            firstRequest.then((page) => {
+                let self = this;
+                let num = page.count;
+                let requests = [firstRequest];
+                if (page.count > pageSize) {
+                    let requestMaker = function *(totalResults) {
+                        let pageNum = 1;
+                        while (pageNum * pageSize <= totalResults) {
+                            let pageParams = {
+                                pageSize: pageSize,
+                                page: pageNum,
+                                sort: 'createdAt,desc'
+                            };
+                            yield self.query(pageParams);
+                            pageNum += 1;
+                        }
+                    };
+
+                    requests = requests.concat(Array.from(requestMaker(num)));
+                }
+
+                this.$q.all(requests).then(
+                    (allResponses) => {
```

Comment:
I'd probably stick an exponential backoff here for failed requests - if one fails because our server is inexplicably under high load, then they won'd necessarily fail, and they should still be able to view pages that were fetched properly

Reply:
Not sure that this would be desirable. I would rather them get no search results than get partial search results.